docs/reference/ingest/ingest-node.asciidoc

```diff
@@ -721,12 +721,29 @@ All processors are defined in the following way within a pipeline definition:
 // NOTCONSOLE
 
 Each processor defines its own configuration parameters, but all processors have
-the ability to declare `tag` and `on_failure` fields. These fields are optional.
+the ability to declare `tag` ,`on_failure` and `if` fields. These fields are optional.
 
 A `tag` is simply a string identifier of the specific instantiation of a certain
 processor in a pipeline. The `tag` field does not affect the processor's behavior,
 but is very useful for bookkeeping and tracing errors to specific processors.
 
+The `if` field must contain a script that returns a boolean value. If the script evaluates to `true`
+then the processor will be executed for the given document otherwise it will be skipped.
+The `if` field takes a map with the script settings used defined in <<script-processor, script-options>>
+and accesses a read only version of the document via the same `ctx` variable used by scripts in the
+<<script-processor>>.
+
+[source,js]
+--------------------------------------------------
+{
+  "bytes": {
+    "if": "ctx.bar == 'expectedValue'",
+    "field": "foo"
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
```

Comment:
Why is this omitted from tests? Can we make an example that will work?

Reply:
I figured just with the example config we don't have enough to go by for running anything, we'd need to actually index a document against a concrete pipeline here to make a test out of it right?
(that may be a little confusing given we only want an example of the `if` field here? Idk, you decide :))

Reply:
IIRC we can have extra setup that is hidden from the generated documentation. I think having the examples always "work" is key to keeping the documentation up to date as apis change.

Reply:
Ok :) Will look into that tomorrow morning :)

Reply:
I can't add a test for a snipped like this it seems. I get this error:

```
Execution failed for task ':docs:buildRestTests'.
> reference/ingest/ingest-node.asciidoc[737:745](js)// CONSOLE: Didn't match (?:(?<comment>#.+)|(?:(?:(?<method>GET|PUT|POST|HEAD|OPTIONS|DELETE)\s+(?<pathAndQuery>[^\n]+)(?<body>(?:\n(?!GET|PUT|POST|HEAD|OPTIONS|DELETE|startyaml|#)[^\n]+)+)?)|(?:startyaml(?s)(?<yaml>.+?)(?-s)endyaml)))\n+: {
    "set": {
      "if": "ctx.bar == 'expectedValue'",
      "field": "foo",
      "value": "bar"
    }
  }
```

if I try anything here. => Seems I at least need a snippet that includes an actual request. That would break with the style of the following docs that also just show a quick outline of the configuration (without tests) for each processor?